src/templates/script/ScriptTemplateProvider.ts

```diff
@@ -58,15 +58,24 @@ export class ScriptTemplateProvider extends TemplateProviderBase {
             const filePath: string = path.join(templatesPath, 'templates.zip');
             await requestUtils.downloadFile(context, templateRelease.templates, filePath);
 
-            await new Promise((resolve: (value?: unknown) => void, reject: (e: Error) => void): void => {
-                // eslint-disable-next-line @typescript-eslint/no-unsafe-call
-                extract(filePath, { dir: templatesPath }, (err: Error) => {
-                    if (err) {
-                        reject(err);
-                    }
-                    resolve();
-                });
-            });
+            await new Promise(
+                (
+                    resolve: (value?: unknown) => void,
+                    reject: (e: Error) => void
+                ): void => {
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+                    extract(filePath, { dir: templatesPath }).then(
+                        () => {
+                            resolve();
+                        },
+                        (err) => {
+                            if (err) {
+                                reject(err);
+                            }
+                        }
+                    )
+                }
+            )
```

Comment:
I think we can switch from a callback to a promise now. All of this would change to one line:
```
await extract(filePath, { dir: templatesPath });
```